src/promptflow/promptflow/core/_utils.py

```diff
@@ -110,6 +121,269 @@ def generate_flow_meta(
     return flow_meta
 
 
+def render_jinja_template_content(template_content, *, trim_blocks=True, keep_trailing_newline=True, **kwargs):
+    template = Template(template_content, trim_blocks=trim_blocks, keep_trailing_newline=keep_trailing_newline)
+    return template.render(**kwargs)
+
+
+def get_connection(connection):
+    if not isinstance(connection, (str, dict)):
+        error_message = """
+        Illegal definition of connection. Need to provide connection name or connection info like below:
```

Comment:
maybe we just say we expect str and dict here?

Reply:
fix it